test/ArchToken.js

```diff
@@ -0,0 +1,68 @@
+// We import Chai to use its asserting functions here.
+const { expect } = require("chai");
+const exp = require("constants");
+const { ethers } = require("hardhat");
+
+describe("Arch Token test suit", function () {
+    let totalSupply;
+    let contract;
+    let token;
+    let owner;
+    let user1;
+    let user2;
+    let users;
+
+    beforeEach(async function () {
+        contract = await ethers.getContractFactory("ArchToken");
+        [owner, user1, user2, ...users] = await ethers.getSigners();
+        token = await contract.deploy();
+        totalSupply = await token.totalSupply();
+    });
+
+    describe("Pre-Mint", function () {
+        it("Should have pre-mint totalSupply of 100m", async function () {
+            // convert from BigNumber to readable value
+            totalSupply = ethers.utils.formatUnits(totalSupply, "ether");
+            // formatUnits() returns a number with the tenths place included
+            expect(totalSupply).to.eq("100000000.0");
+        });
+    });
+
+    // basic end-to-end testing of underlying erc20
+    describe("Transactions", function () {
+        it("Should not be able to transfer() more than total supply", async function () {
+            let amount = totalSupply + 1;
```

Comment:
There is a lower boundary here. It actually reverts when you try to transfer more than `balanceOf`

It worth checking though that `balanceOf` treasury is less or equal to `totalSupply`